bchrpc/server.go

```diff
@@ -983,26 +1039,44 @@ func (s *GrpcServer) GetAddressUnspentOutputs(ctx context.Context, req *pb.GetAd
 		}
 	}
 
+	var tokenMetadata []*pb.TokenMetadata
+	if req.IncludeTokenMetadata && s.slpIndex != nil {
+		tokenMetadata = make([]*pb.TokenMetadata, 0)
+		for _hash := range tokenMetadataSet {
+			tm, _ := s.buildTokenMetadata(_hash)
+			if tm != nil {
+				tokenMetadata = append(tokenMetadata, tm)
+			}
+		}
+	}
+
 	resp := &pb.GetAddressUnspentOutputsResponse{
-		Outputs: utxos,
+		Outputs:       utxos,
+		TokenMetadata: tokenMetadata,
 	}
 	return resp, nil
 }
 
 // GetUnspentOutput takes an unspent output in the utxo set and returns
 // the utxo metadata or not found.
 func (s *GrpcServer) GetUnspentOutput(ctx context.Context, req *pb.GetUnspentOutputRequest) (*pb.GetUnspentOutputResponse, error) {
+
+	if req.IncludeTokenMetadata && s.slpIndex == nil {
+		return nil, status.Error(codes.Unavailable, "slpindex required")
```

Comment:
Nice, love the short circuit pattern here. =)

Reply:
thanks :)